src/Core/CoreMacros.hpp

```diff
@@ -223,7 +223,16 @@
 #   define STRONG_INLINE  inline
 #   define NO_INLINE      __attribute__((noinline))
 
-#   define DEPRECATED __attribute__((deprecated))
+#   undef DEPRECATED
+# if defined(__GNUC__)
+#  if (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__) >= 40500 /* Test for GCC >= 4.5.0 */
+#    define DEPRECATED(msg) __attribute__ ((deprecated(msg)))
+#  else
+#    define DEPRECATED(msg) __attribute__ ((deprecated))
+#  endif
+# elif defined(__clang__)
+#  define DEPRECATED(msg) __attribute__ ((deprecated(msg)))
+# endif

```

Comment:
I believe that if we are to change this we should start using the C++14 `[[deprecated]]` attribute (as you did in the code below) and remove these macros.

Reply:
The point here is I don't want to change the macro, juste make it compatible with OpenMesh one.
Let this modification for next  PR.

Reply:
Yes, we should add an issue to remove them.